### ISSUE TITLE
Token-level loss (DAPO Issue #775) is a no-op — sequence-level normalization used instead

### ISSUE THREAD

**Issue by akaszubski** [akaszubski/autonomous-dev#413]
## Problem

`_compute_grpo_loss` in `src/realign/backends/pytorch/grpo_trainer.py` implements the `token_level_loss` branch (lines 375-379) as:

```python
if self.token_level_loss:
    policy_loss = -policy_loss.sum() / max(policy_loss.numel(), 1)
else:
    policy_loss = -policy_loss.mean()
```

At this point `policy_loss` is a 1D tensor of shape `[batch_size]` — one scalar per sequence — because `_compute_log_probs` has already summed per-token log-probs across the sequence dimension (line 711: `sequence_logps = masked_logps.sum(dim=-1)`).

`tensor.sum() / tensor.numel()` is mathematically identical to `tensor.mean()` for a 1D tensor. The two branches produce identical numerical outputs. Enabling `--token-level-loss` has zero effect on training.

True DAPO token-level loss normalization weights each sequence's contribution by its length relative to the total tokens in the batch: `objective / total_tokens` where `total_tokens = sum(sequence_lengths)`. This requires passing token counts from `_train_step` into `_compute_grpo_loss`.

## Rule Violated

CLAUDE.md testing requirements: "All new code must have tests. Unit tests for core logic." The existing `TestTokenLevelLoss` only checks `trainer.token_level_loss is True` (a flag attribute check), not that the loss value is numerically distinct from the standard path.

## Evidence

- `grpo_trainer.py` lines 375-379: `sum() / numel()` == `mean()` for 1D tensor
- `grpo_trainer.py` lines 694-712: `_compute_log_probs` returns per-sequence scalar via `.sum(dim=-1)`
- `test_grpo_dapo.py` lines 271-283: `TestTokenLevelLoss` only tests flag state, not loss computation

## Suggested Fix

1. Pass response lengths into `_compute_grpo_loss`:
   - Change signature to `_compute_grpo_loss(self, log_probs, ref_log_probs, advantages, response_lengths=None)`
   - In `_train_step`, compute `response_lengths = batch["attention_mask"].sum(dim=-1).float()`
2. In the `token_level_loss` branch, normalize by total tokens:
   ```python
   if self.token_level_loss and response_lengths is not None:
       total_tokens = response_lengths.sum().clamp(min=1)
       # Weight each sequence by its length
       weights = response_lengths / total_tokens
       policy_loss = -(policy_loss * weights).sum()
   ```
3. Add a behavioral test verifying loss values differ between equal-length and unequal-length batches when `token_level_loss=True`.

---
*Filed automatically by continuous-improvement-analyst*
